### PR TITLE
Ruby debugging lesson: VSCode update

### LINKED ISSUE #23633: Ruby Debugging: Update Lesson
- [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
- [x] The title of this issue follows the `where the change occurs: brief description of request` format, e.g. `Global SCSS file: Buttons should have XYZ`
- [x] I would like to be assigned this issue to work on it

**1. Description of the Feature Request**
This feature will update the Ruby Debugging lesson to include the VSCode integrated Ruby debugger. 

This update would make debugging Ruby projects quicker, and simpler through the use of a new debugging tool. 

This should help reduce the number of questions in the #ruby-help channel, by making it easier for learners to debug whichever projects they struggle with the most. 

**2. Additional Information**
I think this can be a short addition towards the very end of the [Debugging lesson](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/debugging). 
Since this is already a lengthy lesson, and learners got to learn about concepts such as Watch, Scope in the Foundations [Javascript Developer Tools lesson](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/javascript-developer-tools), my thoughts are that the new information can be concise.

At a minimum I was thinking this addition could include: 
1) How to install the necessary gems and set up VSCode Ruby Debugger
2) How to set breakpoints and watch a variable. To save space this can even be done using the last example in the lesson's pry-byebug section. 
3) A brief mention regarding using `binding.break` to run pry *within* the VSCode debugger.

### PR DESCRIPTION
**Pull request by mgrigoriev8109**
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

**1. Because:**
Closes #23633

**2. This PR:**
Is a work in progress for updating the current [Ruby Debugging lesson](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/debugging) to include the VSCode Debugging tool. Due to the importance of the concepts covered in the beginning of the lesson (stack trace, using `p`, using pry) I didn't touch any of that content.

The content I added is all in the Assignments, and focused on:
1) Guiding learners to install the VSCode debugger
2) Editing their launch.json Configuration 
3) Encouraging them to play with the VSCode Debugger and see the similarities to the Dev Tools debugger they already learned about
4) Pointing out the similar functionality of `binding.break` to `binding.pry`, and the importance of practicing using `pry` even though the VSCode Debugger exists

### CODE REVIEW

**Review by zachmmeyer — APPROVED** [review]
This is really great! Thank you for doing this!

**Comment by mgrigoriev8109** [comment]
I'm actually wondering if telling learners about `binding.break` is a bit much all at once, since the lesson literally just introduced `pry-byebug` and no one has really gotten a chance to practice that in their projects.

So I wonder if I should remove the mention of `binding.break`, and in the **Assignment** section stick to stating that while you *can* use the VSCode Debugger, you should still practice using `pry-byebug` on its own due to its current popularity as a Debugging tool.

**Comment by rlmoser99** [comment]
@mgrigoriev8109 Thank you for taking the initiative to work on this issue. I will take a closer look at this tomorrow, but in reading your note, I am reminded that we would like to create exercises that gives people examples to practice debugging. 

https://github.com/TheOdinProject/ruby-exercises/issues/15

These do not need to have tests, like the other ones in ruby-exercises, but basically some methods with built in bugs. I am wondering if we should consider exercises to go along with this lesson before we merge in these updates. 

Would you be interested in coming up with examples? The trickiest part would be coming up with examples without using enumerables, since those lessons are later. Maybe we could even use some of the examples from the array, hash, and method exercises.

**Comment by mgrigoriev8109** [comment]
@rlmoser99 I would definitely be interested in creating some exercises for this lesson to practice debugging. If it would be helpful I can also include tests - since learners will already be familiar with that format.

**Comment by mgrigoriev8109** [comment]
Just now updated the Assignment while checking all the instructions on a newly installed VM, everything worked for me getting the VSCode debugger running but I'd love input on how I can make things easier to understand. 

I'll now be working on the debugging exercises that I'll insert into the end of this lesson as practice, and should have a PR ready for https://github.com/TheOdinProject/ruby-exercises/issues/15 by the end of the week. I'm going to create a couple exercises instructing learners to rely on the stack trace, a couple instructing them to rely on `puts`, a couple instructing them to rely on `pry-byebug`, and a couple instructing them to use the VSCode debugger GUI.

File: ruby_programming/basic_ruby/debugging.md [hunk 1]
```diff
@@ -197,8 +197,14 @@ Obviously, if available, <span id='debugging-with-stack-trace'>the stack trace i
...
+4. If you're having a hard time figuring out how to navigate to your launch.json file in order to change the Configuration, read through this in-depth guide on [Debugging with VScode](https://code.visualstudio.com/docs/editor/debugging). 
+5. Check your [VSCode Extensions](https://code.visualstudio.com/docs/editor/extension-marketplace) and make sure the [Ruby Extension](https://marketplace.visualstudio.com/items?itemName=rebornix.Ruby) and the [VSCode rbdg Ruby Debugger](https://marketplace.visualstudio.com/items?itemName=KoichiSasada.vscode-rdbg) extensions are installed.
+6. Now that everything is installed, configured, let's create a new file which you can call `script.rb`. Next copy and paste the very first example in the<a href='#debugging-with-pry-byebug'>Debugging with pry-byebug</a> Section. So include everything from `require 'pry-byebug'` to `isogram?("Odin")` in our new file. Save the file.
```
Comment (zachmmeyer) [inline ruby_programming/basic_ruby/debugging.md:205]:
```suggestion
6. Now that everything is installed, configured, let's create a new file which you can call `script.rb`. Next copy and paste the very first example in the <a href='#debugging-with-pry-byebug'>Debugging with pry-byebug</a> Section. So include everything from `require 'pry-byebug'` to `isogram?("Odin")` in our new file. Save the file. 
```

**Review by zachmmeyer — CHANGES_REQUESTED** [review]
I found like one typo.
I followed the instructions and it works perfectly. Really awesome. Excited for the exercises!

**Comment by zachmmeyer** [comment]
@mgrigoriev8109 Let me know when we're good!

**Comment by mgrigoriev8109** [comment]
@zachmmeyer I created a pull request for the Ruby Debugging exercises: https://github.com/TheOdinProject/ruby-exercises/pull/69

, so once that pull request is reviewed and approved (Or any requested changes are made) this lesson should be all ready to go!

**Comment by zachmmeyer** [comment]
@mgrigoriev8109 Understood!

**Pull request merged**